src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenRefReturnTests.cs

```diff
@@ -3590,5 +3590,95 @@ static void Main()
 Program+RefFunc1`2[Derived1,Base]", verify: Verification.Passes);
         }
 
+        [WorkItem(25024, "https://github.com/dotnet/roslyn/issues/25024")]
+        [Fact]
+        public void RefReturnDiscardLifetime()
+        {
+            var text = @"
+    class Program
+    {
+        static bool flag = true;
+
+        public static void Main()
+        {
+            if (flag)
+            {
+                ref var local1 = ref M1(out var _);
+                ref var local2 = ref M1(out var _);
+
+                local1 = 1;
+                local2 = 2;
+                System.Console.Write(local1 + local2);
+            }
+
+            if (flag)
+            {
+                ref var local1 = ref M1(out var _);
+                ref var local2 = ref M1(out var _);
+
+                local1 = 3;
```

Comment:
I didn't understand the purpose of the second block.

Reply:
In theory we could reuse "leaked" temps from different blocks, but we don't (for simplicity).
I have added a second block just to see whether such reuse happens or not.